tests/Integration/Behaviour/Features/Context/Domain/CartFeatureContext.php

```diff
@@ -146,17 +147,22 @@ public function addProductsToCart(int $quantity, string $productName, string $ca
     {
         $productId = $this->getProductIdByName($productName);
 
-        $this->getCommandBus()->handle(
-            new UpdateProductQuantityInCartCommand(
-                SharedStorage::getStorage()->get($cartReference),
-                $productId,
-                $quantity
-            )
-        );
-        SharedStorage::getStorage()->set($productName, $productId);
+        $this->lastException = null;
+        try {
+            $this->getCommandBus()->handle(
+                new UpdateProductQuantityInCartCommand(
+                    SharedStorage::getStorage()->get($cartReference),
+                    $productId,
+                    $quantity
+                )
+            );
+            SharedStorage::getStorage()->set($productName, $productId);
 
-        // Clear cart static cache or it will have no products in next calls
-        Cart::resetStaticCache();
+            // Clear cart static cache or it will have no products in next calls
+            Cart::resetStaticCache();
+        } catch (CartException $e) {
```

Comment:
```suggestion
        } catch (MinimalQuantityException $e) {
```

We try not to use wide exceptions, so that unexpected bugs are blocking right away So prefer using the most specific exception

Reply:
Fixed